mp/src/game/shared/momentum/mom_stickybomb.cpp

```diff
@@ -0,0 +1,385 @@
+
+        rotInterpolator.AddToHead(changeTime, &vCurAngles, false);
+    }
+}
+
+int CMomStickybomb::DrawModel(int flags)
+{
+    // Don't draw the stickybomb during the first 0.1 seconds.
+    if (gpGlobals->curtime - m_flSpawnTime < 0.1f)
+    {
+        return 0;
+    }
+
+    return BaseClass::DrawModel(flags);
+}
+
+void CMomStickybomb::Spawn()
+{
+    m_flSpawnTime = gpGlobals->curtime;
+    BaseClass::Spawn();
+}
+
+void CMomStickybomb::Pulse()
+{
+    if (m_bPulsed == false)
+    {
+        if ((gpGlobals->curtime - m_flCreationTime) >= 0.8f)
+        {
+            ParticleProp()->Create("stickybomb_pulse_red", PATTACH_ABSORIGIN);
+            m_bPulsed = true;
+        }
+    }
+}
+
+#else
+
+void CMomStickybomb::Spawn()
+{
+    BaseClass::Spawn();
+
+    UseClientSideAnimation();
+    SetCollisionGroup(COLLISION_GROUP_PROJECTILE);
+    SetSolidFlags(FSOLID_NOT_STANDABLE);
+    SetMoveType(MOVETYPE_FLYGRAVITY, MOVECOLLIDE_FLY_BOUNCE);
+    SetSolid(SOLID_BBOX);
+    AddEffects(EF_NOSHADOW);
+    SetSize(Vector(-2, -2, -2), Vector(2, 2, 2));
+    AddFlag(FL_GRENADE);
+
+    VPhysicsInitNormal(SOLID_BBOX, 0, false);
+
+    m_bTouched = false;
+    m_flCreationTime = gpGlobals->curtime;
+    m_takedamage = DAMAGE_NO;
+    SetGravity(MOM_STICKYBOMB_GRAVITY);
+    SetFriction(MOM_STICKYBOMB_FRICTION);
+    SetElasticity(MOM_STICKYBOMB_ELASTICITY);
+
+    SetTouch(&CMomStickybomb::StickybombTouch);
+    SetThink(&CMomStickybomb::StickybombThink);
+    SetNextThink(gpGlobals->curtime + 0.2);
+}
+
+void CMomStickybomb::Precache()
+{
+    BaseClass::Precache();
+    PrecacheModel("models/weapons/w_models/w_stickybomb.mdl");
+    PrecacheParticleSystem("stickybombtrail_red");
+    PrecacheParticleSystem("stickybomb_pulse_red");
+    // PrecacheScriptSound("Stickybomb.Bounce");
+}
+
+void CMomStickybomb::SetupInitialTransmittedGrenadeVelocity(const Vector &velocity) { m_vInitialVelocity = velocity; }
+
+CMomStickybomb *CMomStickybomb::Create(const Vector &position, const QAngle &angles, const Vector &velocity,
+                                       const AngularImpulse &angVelocity, CBaseCombatCharacter *pOwner)
+{
+    CMomStickybomb *pStickybomb =
+        static_cast<CMomStickybomb *>(CBaseEntity::CreateNoSpawn("momentum_stickybomb", position, angles, pOwner));
+
+    if (pStickybomb)
+    {
+        pStickybomb->SetModel("models/weapons/w_models/w_stickybomb.mdl");
+        DispatchSpawn(pStickybomb);
+        pStickybomb->InitStickybomb(velocity, angVelocity, pOwner);
+        pStickybomb->ApplyLocalAngularVelocityImpulse(angVelocity);
+        pStickybomb->SetAbsVelocity(velocity);
+        pStickybomb->SetThrower(pOwner);
+        pStickybomb->SetAbsAngles(angles);
+
+        pStickybomb->SetDamage(60.0f);
+        // NOTE: Rocket/Stickybomb explosion radius is 146.0f in TF2, but 121.0f is used for self damage (see
+        // RadiusDamage in mom_gamerules)
+        pStickybomb->SetRadius(146.0f);
```

Comment:
Radius gets set here and in line 372 in game rules, is there a reason for that?

Reply:
It seems that not setting the radius here results in incorrect push force calculation, so I'm keeping it here for now.